### PR TITLE
Add unit test for GovernanceRepositoryAPIImpl

### LINKED ISSUE #28545: Add unit test for GovernanceRepositoryAPIImpl persistLatestCheckJobId, getLatestCheckJobId, deleteLatestCheckJobId
Hi community,

This issue is for https://github.com/apache/shardingsphere/discussions/28487.

## Aim
Add more unit tests for `GovernanceRepositoryAPIImpl` to test its `persistLatestCheckJobId`, `getLatestCheckJobId` and `deleteLatestCheckJobId` functions to improve unit test coverage. Unit test could be added in existing `GovernanceRepositoryAPIImplTest`.

Unit test coverage can be viewed at: https://app.codecov.io/gh/apache/shardingsphere

## Basic Qualifications
- Java
- Maven
- Junit

### PR DESCRIPTION
**Pull request by ManaMatrix**
Fixes #28545.

Changes proposed in this pull request:

-  Added unit tests for persistLatestCheckJobId
-  Added unit tests for getLatestCheckJobId
-  Added unit tests for deleteLatestCheckJobId

---

Before committing this PR, I'm sure that I have checked the following options:
- [x] My code follows the [code of conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/) of this project.
- [x] I have self-reviewed the commit code.
- [x] I have (or in comment I request) added corresponding labels for the pull request.
- [x] I have passed maven check locally : `./mvnw clean install -B -T1C -Dmaven.javadoc.skip -Dmaven.jacoco.skip -e`.
- [x] I have made corresponding changes to the documentation.
- [x] I have added corresponding unit tests for my changes.

### CODE REVIEW

File: test/it/pipeline/src/test/java/org/apache/shardingsphere/test/it/data/pipeline/core/job/service/GovernanceRepositoryAPIImplTest.java [hunk 1]
```diff
@@ -48,9 +48,7 @@
 import static org.hamcrest.CoreMatchers.anyOf;
 import static org.hamcrest.CoreMatchers.is;
 import static org.hamcrest.MatcherAssert.assertThat;
-import static org.junit.jupiter.api.Assertions.assertFalse;
-import static org.junit.jupiter.api.Assertions.assertNotNull;
-import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.*;
```
Comment (sandynz) [inline test/it/pipeline/src/test/java/org/apache/shardingsphere/test/it/data/pipeline/core/job/service/GovernanceRepositoryAPIImplTest.java:51]: The importing could not be combined. Suggest to import ShardingSphere code style in IDEA, path: `src/resources/code-style-idea.xml`

File: test/it/pipeline/src/test/java/org/apache/shardingsphere/test/it/data/pipeline/core/job/service/GovernanceRepositoryAPIImplTest.java [hunk 2]
```diff
@@ -155,7 +153,39 @@ void assertPersistJobOffsetInfo() {
...
+
+    @Test
+    void assertDeleteLatestCheckJobId() {
+        String parentJobId = "parentJob123";
+        String checkJobId = "checkJob456";
+        governanceRepositoryAPI.persistLatestCheckJobId(parentJobId, checkJobId);
+        assertTrue(governanceRepositoryAPI.getLatestCheckJobId(parentJobId).isPresent(), "Expected checkJobId to be present before deletion");
+        governanceRepositoryAPI.deleteLatestCheckJobId(parentJobId);
+        assertFalse(governanceRepositoryAPI.getLatestCheckJobId(parentJobId).isPresent(), "Expected no checkJobId to be present after deletion");
+    }
```
Comment (sandynz) [inline test/it/pipeline/src/test/java/org/apache/shardingsphere/test/it/data/pipeline/core/job/service/GovernanceRepositoryAPIImplTest.java:187]: It's better to combine them in one unit test method, in order to reduce duplicated code.

**Review by sandynz — CHANGES_REQUESTED** [review]

File: test/it/pipeline/src/test/java/org/apache/shardingsphere/test/it/data/pipeline/core/job/service/GovernanceRepositoryAPIImplTest.java (hunk 2 above)
Comment (ManaMatrix) [reply test/it/pipeline/src/test/java/org/apache/shardingsphere/test/it/data/pipeline/core/job/service/GovernanceRepositoryAPIImplTest.java]: It makes sense. Just combine them in one unit test method.

File: test/it/pipeline/src/test/java/org/apache/shardingsphere/test/it/data/pipeline/core/job/service/GovernanceRepositoryAPIImplTest.java [hunk 3]
```diff
@@ -155,7 +156,26 @@ void assertPersistJobOffsetInfo() {
...
         assertThat(actual.get(), is("testValue"));
     }
-    
+
+    @Test
+    void testGovernanceRepositoryAPIImplMethods() {
+        String parentJobId1 = "parentJob123";
+        String checkJobId1 = "checkJob456";
+        governanceRepositoryAPI.persistLatestCheckJobId(parentJobId1, checkJobId1);
+        Optional<String> retrievedCheckJobId1 = governanceRepositoryAPI.getLatestCheckJobId(parentJobId1);
```
Comment (sandynz) [inline test/it/pipeline/src/test/java/org/apache/shardingsphere/test/it/data/pipeline/core/job/service/GovernanceRepositoryAPIImplTest.java:165]: From Code of Conduct, `retrievedCheckJobId1` could be `actualCheckJobId`

**Review by sandynz — CHANGES_REQUESTED** [review]

File: test/it/pipeline/src/test/java/org/apache/shardingsphere/test/it/data/pipeline/core/job/service/GovernanceRepositoryAPIImplTest.java (hunk 3 above)
Comment (ManaMatrix) [reply test/it/pipeline/src/test/java/org/apache/shardingsphere/test/it/data/pipeline/core/job/service/GovernanceRepositoryAPIImplTest.java]: Solved!

File: test/it/pipeline/src/test/java/org/apache/shardingsphere/test/it/data/pipeline/core/job/service/GovernanceRepositoryAPIImplTest.java [hunk 4]
```diff
@@ -155,7 +156,26 @@ void assertPersistJobOffsetInfo() {
...
+        assertTrue(retrievedCheckJobId1.isPresent(), "Expected a checkJobId to be present");
+        assertEquals(retrievedCheckJobId1.get(), checkJobId1, "The retrieved checkJobId does not match the persisted one");
+        String parentJobId2 = "parentJob789";
+        String checkJobId2 = "checkJob012";
+        governanceRepositoryAPI.persistLatestCheckJobId(parentJobId2, checkJobId2);
+        Optional<String> retrievedCheckJobId2 = governanceRepositoryAPI.getLatestCheckJobId(parentJobId2);
+        assertTrue(retrievedCheckJobId2.isPresent(), "Expected a checkJobId to be present");
+        assertEquals(checkJobId2, retrievedCheckJobId2.get(), "The retrieved checkJobId does not match the expected one");
+        governanceRepositoryAPI.deleteLatestCheckJobId(parentJobId2);
+        assertFalse(governanceRepositoryAPI.getLatestCheckJobId(parentJobId2).isPresent(), "Expected no checkJobId to be present after deletion");
```
[the remark this answers is not in the document]
Comment (ManaMatrix) [reply test/it/pipeline/src/test/java/org/apache/shardingsphere/test/it/data/pipeline/core/job/service/GovernanceRepositoryAPIImplTest.java]: Solved!

**Comment by sandynz** [comment]
Hi @ManaMatrix , there's Spotless error in CI, you could follow error message to fix it.

**Review by sandynz — APPROVED** [review]

**Comment by sandynz** [comment]
Nice. Merged

**Pull request merged**